hertzbeat-warehouse/src/main/java/org/apache/hertzbeat/warehouse/store/DataStorageDispatch.java

```diff
@@ -87,24 +91,16 @@ protected void calculateMonitorStatus(CollectRep.MetricsData metricsData) {
         if (metricsData.getPriority() == 0) {
             long id = metricsData.getId();
             CollectRep.Code code = metricsData.getCode();
-            // query current status
-            String queryStatusSql = "SELECT status FROM hzb_monitor WHERE id = ?";
             try {
-                int currentStatus = jdbcTemplate.queryForObject(queryStatusSql, Integer.class, id);
-                if (code == CollectRep.Code.SUCCESS && currentStatus == CommonConstants.MONITOR_DOWN_CODE) {
-                    // if collect success and current status is DOWN, update to UP
-                    String sql = "UPDATE hzb_monitor SET status = ? WHERE id = ?";
-                    jdbcTemplate.update(sql, CommonConstants.MONITOR_UP_CODE, id);
-                } else if (code != CollectRep.Code.SUCCESS && currentStatus == CommonConstants.MONITOR_UP_CODE) {
-                    // if collect failed and current status is UP, update to DOWN
-                    String sql = "UPDATE hzb_monitor SET status = ? WHERE id = ?";
-                    jdbcTemplate.update(sql, CommonConstants.MONITOR_DOWN_CODE, id);
+                String sql = "UPDATE hzb_monitor SET status = ? WHERE id = ? AND status != ?";
+                int status = code == CollectRep.Code.SUCCESS ? CommonConstants.MONITOR_UP_CODE : CommonConstants.MONITOR_DOWN_CODE;
+                int matchedRows = jdbcTemplate.update(sql, status, id, status);
```

Comment:
👍 hi, how about this? `status=?` can use the database index, and ignore case when status is PAUSED
```suggestion
                String sql = "UPDATE hzb_monitor SET status = ? WHERE id = ? AND status = ?";
                int status = code == CollectRep.Code.SUCCESS ? CommonConstants.MONITOR_UP_CODE : CommonConstants.MONITOR_DOWN_CODE;
                int preStatus = code == CollectRep.Code.SUCCESS ? CommonConstants.MONITOR_DOWN_CODE : CommonConstants.MONITOR_UP_CODE;
                int matchedRows = jdbcTemplate.update(sql, status, id, preStatus);
```

Reply:
👍hi, I think it's very good